tests/testutils.py

```diff
@@ -0,0 +1,64 @@
+"""Test utilities
+
+This module includes various methods that help in writing the tests.
+It should not be imported by any module except for test code.
+"""
+
+import time
+import os
+
+
+def modify_test_reload_module_file(shared_datadir):
+    """This method modifies the content of the test_reload.py module"""
+    module = shared_datadir.joinpath('test_reload.py')
+
+    module.write_text("""provides = ['get_message']
+def get_message():
+    return 'Hello World'""")
+
+
+def modify_with_dependency_module_file(shared_datadir):
+    """This method modifies the content of the with_dependency.py module"""
+    module = shared_datadir.joinpath('with_dependency.py')
+
+    module.write_text("""requires = ['test_reload']
+provides = ['generate_message']
+
+def generate_message():
+    return 'Message: ' + get_message() + ' - ' + get_message()""")
+
+
+def await_queue_size(file_watcher, expected_queue_size):
+    """
+    This method Waits for the size of the queue to reach the given expected queue
+    size number. The loop exists if the number is not reached after 10 seconds.
+    """
+
+    for _ in range(30, 0, -1):
+        if file_watcher.modified_files_queue.qsize() == expected_queue_size:
+            break
+        time.sleep(0.5)
+
+
+def get_last_modified_file_time(path):
+    """This method gets the time that the given file was last modified.
+    :param path: the path to the file
+    """
+    return os.stat(path).st_mtime
+
+
+def was_file_modified(path, last_modified_time):
```

Comment:
Modified files that are not watched are not added to the queue, so there are tests that test what happens when non-watched files are modified but auto-reload is on. We still have to check that the file was modified hence the reason why this method was introduced.